File: backend/backend-messages/TestData.cs
using Microsoft.EntityFrameworkCore;
using backend_messages.Data;
using backend_messages.Models;

namespace backend_messages
{
    public static class TestData
    {
        public static async Task SeedTestDataAsync(MessageDbContext context)
        {
            if (await context.Conversations.AnyAsync())
            {
                return; 
            }

            var conv1 = new Conversation
            {
                Id = Guid.NewGuid(),
                User1Id = Guid.Parse("11111111-1111-1111-1111-111111111111"),
                User2Id = Guid.Parse("22222222-2222-2222-2222-222222222222"), 
                CreatedAt = DateTime.UtcNow.AddDays(-2),
                UpdatedAt = DateTime.UtcNow.AddMinutes(-2)
            };

            var conv2 = new Conversation
            {
                Id = Guid.NewGuid(),
                User1Id = Guid.Parse("11111111-1111-1111-1111-111111111111"), 
                User2Id = Guid.Parse("33333333-3333-3333-3333-333333333333"), 
                CreatedAt = DateTime.UtcNow.AddDays(-1),
                UpdatedAt = DateTime.UtcNow.AddHours(-1)
            };

            await context.Conversations.AddRangeAsync(conv1, conv2);

            var messages1 = new[]
            {
                new Message
                {
                    Id = Guid.NewGuid(),
                    ConversationId = conv1.Id,
                    SenderId = Guid.Parse("22222222-2222-2222-2222-222222222222"),
                    Content = "Hi! I saw your portfolio and I'm really impressed with your projects!",
                    MessageType = "Text",
                    CreatedAt = DateTime.UtcNow.AddHours(-4),
                    UpdatedAt = DateTime.UtcNow.AddHours(-4)
                },
                new Message
                {
                    Id = Guid.NewGuid(),
                    ConversationId = conv1.Id,
                    SenderId = Guid.Parse("11111111-1111-1111-1111-111111111111"),
                    Content = "Thank you! I really appreciate the feedback.",
                    MessageType = "Text",
                    CreatedAt = DateTime.UtcNow.AddHours(-3),
                    UpdatedAt = DateTime.UtcNow.AddHours(-3)
                },
                new Message
                {
                    Id = Guid.NewGuid(),
                    ConversationId = conv1.Id,
                    SenderId = Guid.Parse("22222222-2222-2222-2222-222222222222"),
                    Content = "Would you be interested in collaborating on a project?",
                    MessageType = "Text",
                    CreatedAt = DateTime.UtcNow.AddMinutes(-2),
                    UpdatedAt = DateTime.UtcNow.AddMinutes(-2)
                }
            };

            var messages2 = new[]
            {
                new Message
                {
                    Id = Guid.NewGuid(),
                    ConversationId = conv2.Id,
                    SenderId = Guid.Parse("33333333-3333-3333-3333-333333333333"),
                    Content = "Hey! Just saw your latest ML project on GitHub. Really clean implementation!",
                    MessageType = "Text",
                    CreatedAt = DateTime.UtcNow.AddHours(-2),
                    UpdatedAt = DateTime.UtcNow.AddHours(-2)
                },
                new Message
                {
                    Id = Guid.NewGuid(),
                    ConversationId = conv2.Id,
                    SenderId = Guid.Parse("11111111-1111-1111-1111-111111111111"),
                    Content = "Thanks Mike! I spent a lot of time optimizing the training pipeline.",
                    MessageType = "Text",
                    CreatedAt = DateTime.UtcNow.AddHours(-1),
                    UpdatedAt = DateTime.UtcNow.AddHours(-1)
                }
            };

            await context.Messages.AddRangeAsync(messages1);
            await context.Messages.AddRangeAsync(messages2);

            await context.SaveChangesAsync();
        }
    }
} 

File: frontend/lib/messages/test-data.ts
// Test data utilities for development and testing


import { MessageResponse, ConversationResponse } from './api';

// Mock conversations for development
export const mockConversations: ConversationResponse[] = [
  {
    id: "conv-1",
    otherUserId: "user-2",
    otherUserName: "Sarah Wilson",
    otherUserAvatar: "https://placehold.co/40x40",
    lastMessage: {
      id: "msg-5",
      senderId: "user-2",
      receiverId: "current-user",
      content: "Thanks for checking out my portfolio!",
      createdAt: new Date(Date.now() - 2 * 60 * 1000).toISOString(), // 2 minutes ago
      conversationId: "conv-1",
      isRead: false
    },
    unreadCount: 1,
    updatedAt: new Date(Date.now() - 2 * 60 * 1000).toISOString(),
    isOnline: true
  },
  {
    id: "conv-2", 
    otherUserId: "user-3",
    otherUserName: "Mike Chen",
    otherUserAvatar: "https://placehold.co/40x40",
    lastMessage: {
      id: "msg-8",
      senderId: "user-3",
      receiverId: "current-user", 
      content: "Great work on the ML project! 🔥",
      createdAt: new Date(Date.now() - 60 * 60 * 1000).toISOString(), // 1 hour ago
      conversationId: "conv-2",
      isRead: true
    },
    unreadCount: 0,
    updatedAt: new Date(Date.now() - 60 * 60 * 1000).toISOString(),
    isOnline: true
  },
  {
    id: "conv-3",
    otherUserId: "user-4", 
    otherUserName: "Alex Johnson",
    otherUserAvatar: "https://placehold.co/40x40",
    lastMessage: {
      id: "msg-11",
      senderId: "current-user",
      receiverId: "user-4",
      content: "That sounds interesting! What kind of startup are you thinking about?",
      createdAt: new Date(Date.now() - 3 * 60 * 60 * 1000).toISOString(), // 3 hours ago
      conversationId: "conv-3",
      isRead: true
    },
    unreadCount: 0,
    updatedAt: new Date(Date.now() - 3 * 60 * 60 * 1000).toISOString(),
    isOnline: false
  }
];

// Mock messages for each conversation
export const mockMessages: Record<string, MessageResponse[]> = {
  "conv-1": [
    {
      id: "msg-1",
      senderId: "user-2",
      receiverId: "current-user",
      content: "Hi! I saw your portfolio and I'm really impressed with your full-stack projects. The e-commerce platform looks amazing!",
      createdAt: new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString(), // Yesterday
      conversationId: "conv-1",
      isRead: true
    },
    {
      id: "msg-2", 
      senderId: "current-user",
      receiverId: "user-2",
      content: "Thank you so much! I really appreciate the feedback. I checked out your design portfolio too - your UI work is incredible!",
      createdAt: new Date(Date.now() - 23 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-1",
      isRead: true
    },
    {
      id: "msg-3",
      senderId: "user-2", 
      receiverId: "current-user",
      content: "Thanks! I was wondering if you'd be interested in collaborating on a project? I have a client who needs both design and development work.",
      createdAt: new Date(Date.now() - 4 * 60 * 60 * 1000).toISOString(), // 4 hours ago
      conversationId: "conv-1",
      isRead: true
    },
    {
      id: "msg-4",
      senderId: "current-user",
      receiverId: "user-2", 
      content: "That sounds really interesting! I'd love to hear more about it. What kind of project is it?",
      createdAt: new Date(Date.now() - 3 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-1", 
      isRead: true
    },
    {
      id: "msg-5",
      senderId: "user-2",
      receiverId: "current-user",
      content: "Thanks for checking out my portfolio!",
      createdAt: new Date(Date.now() - 2 * 60 * 1000).toISOString(),
      conversationId: "conv-1",
      isRead: false
    }
  ],
  "conv-2": [
    {
      id: "msg-6",
      senderId: "user-3",
      receiverId: "current-user", 
      content: "Hey! Just saw your latest ML project on GitHub. The neural network implementation is really clean!",
      createdAt: new Date(Date.now() - 5 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-2",
      isRead: true
    },
    {
      id: "msg-7",
      senderId: "current-user",
      receiverId: "user-3",
      content: "Thanks Mike! I spent a lot of time optimizing the training pipeline. How's your computer vision project going?", 
      createdAt: new Date(Date.now() - 4 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-2",
      isRead: true
    },
    {
      id: "msg-8",
      senderId: "user-3",
      receiverId: "current-user",
      content: "Great work on the ML project! 🔥",
      createdAt: new Date(Date.now() - 60 * 60 * 1000).toISOString(),
      conversationId: "conv-2", 
      isRead: true
    }
  ],
  "conv-3": [
    {
      id: "msg-9",
      senderId: "user-4",
      receiverId: "current-user",
      content: "Would love to collaborate on a project! I have some ideas for a new startup.",
      createdAt: new Date(Date.now() - 6 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-3",
      isRead: true
    },
    {
      id: "msg-10",
      senderId: "current-user", 
      receiverId: "user-4",
      content: "That sounds interesting! What kind of startup are you thinking about?",
      createdAt: new Date(Date.now() - 3 * 60 * 60 * 1000).toISOString(),
      conversationId: "conv-3",
      isRead: true
    }
  ]
};

// Helper function to get mock data (useful during development)
export function getMockConversations(): ConversationResponse[] {
  return mockConversations;
}

export function getMockMessages(conversationId: string): MessageResponse[] {
  return mockMessages[conversationId] || [];
} 